s3/src/main/scala/monix/connect/s3/MultipartUploadSubscriber.scala

```diff
@@ -0,0 +1,185 @@
+  ssekmsEncryptionContext: Option[String],
+  ssekmsKeyId: Option[String],
+  requestPayer: Option[String])(
+  implicit
+  s3Client: S3AsyncClient)
+  extends Consumer.Sync[Array[Byte], CompleteMultipartUploadResponse] {
+
+  private val createRequest: CreateMultipartUploadRequest =
+    S3RequestBuilder.createMultipartUploadRequest(
+      bucket = bucket,
+      key = key,
+      contentType = contentType,
+      acl = acl,
+      grantFullControl = grantFullControl,
+      grantRead = grantRead,
+      grantReadACP = grantReadACP,
+      grantWriteACP = grantWriteACP,
+      requestPayer = requestPayer,
+      serverSideEncryption = serverSideEncryption,
+      sseCustomerAlgorithm = sseCustomerAlgorithm,
+      sseCustomerKey = sseCustomerKey,
+      sseCustomerKeyMD5 = sseCustomerKeyMD5,
+      ssekmsEncryptionContext = ssekmsEncryptionContext,
+      ssekmsKeyId = ssekmsKeyId
+    )
+
+  private val uploadId: Task[String] =
+    Task
+      .from(s3Client.createMultipartUpload(createRequest))
+      .map(_.uploadId())
+      .memoize
+
+  private var buffer: Array[Byte] = Array.emptyByteArray
+
+  private var completedParts: List[CompletedPart] = List.empty[CompletedPart]
+
+  private var partN = 1
+  def createSubscriber(
+    callback: Callback[Throwable, CompleteMultipartUploadResponse],
+    s: Scheduler): (Subscriber.Sync[Array[Byte]], AssignableCancelable) = {
+    val out = new Subscriber.Sync[Array[Byte]] {
+      implicit val scheduler = s
+
+      /**
+        *
+        * If the chunk lenght is bigger than the minimum size, perfom the part request [[UploadPartRequest]],
+        * which returns a [[CompletedPart]].
+        * In the case it is not, the chunk is added to the buffer waiting to be aggregated with the next one.
+        */
+      def onNext(chunk: Array[Byte]): Ack = {
+        if (chunk.length < minChunkSize) {
+          buffer = buffer ++ chunk
+        } else {
+          try {
+            uploadPart(bucket, key, partN, uploadId, buffer)
+          } catch {
+            case ex if NonFatal(ex) => onError(ex)
+          }
+          partN += 1
+          buffer = Array.emptyByteArray
+        }
+        monix.execution.Ack.Continue
+      }
+
+      /**
+        * Checks that the buffer is not empty and performs the last part request [[CompletedPart]] of the buffer in case it was not.
+        * Finally it creates a request to indicate that the multipart upload was completed [[CompleteMultipartUploadRequest]],
+        * returning a [[CompleteMultipartUploadResponse]] that is passed to the callback.
+        */
+      def onComplete(): Unit = {
+        if (!buffer.isEmpty) {
+          try {
+            uploadPart(bucket, key, partN, uploadId, buffer)
+          } catch {
+            case ex if NonFatal(ex) => onError(ex)
+          }
+        }
+        val response: Task[CompleteMultipartUploadResponse] = for {
+          uid            <- uploadId
+          completedParts <- Task(completedParts)
```

Comment:
Can `completedParts` be accessed directly?

Reply:
yup 👍